script.js

```diff
@@ -1,12 +1,194 @@
+  header.appendChild(navigationContainer);
+  navigationContainer.id = "navigation-container";
+  navigationContainer.className = "navigation-container";
+
+  let selector = document.createElement("select");
+  navigationContainer.appendChild(selector);
+  selector.id = "select-menu";
+  selector.className = "select-menu";
+  selector.name = "select-menu";
+  selector.ariaLabel = "Select episode";
+  renderEpisodeSelectorList();
+
+  let searchBox = document.createElement("input");
+  navigationContainer.appendChild(searchBox);
+  searchBox.id = "search-box";
+  searchBox.className = "search-box";
+  searchBox.type = "text";
+  selector.ariaLabel = "search";
+  searchBox.placeholder = " Episode search...";
+
+  let searchEpisodes = document.querySelector("#search-box");
+  searchEpisodes.addEventListener("keyup", () => {
+    searchAllEpisodes();
+  });
+
+  let displayNumberOfEpisodes = document.createElement("span");
+  navigationContainer.appendChild(displayNumberOfEpisodes);
+  displayNumberOfEpisodes.id = "display-status";
+  displayNumberOfEpisodes.className = "display-status";
+  renderNumberOfEpisodes(allEpisodes.length, allEpisodes.length);
+};
+
+// Dynamically renders episodes in selector list and adds an event listener to the `select` element.
+// Originally I attached a "click" listener to each `option` using the loop which worked in Firefox but it didn't work in chrome
+// so the workaround I found was to use "change" on the `select` element and pass the `selector.value` instead of the `episode.name`
+// As the `selector.value` is equal to `episode.name` I think(hope) this is ok to pull the data from the DOM and not directly from the object???
+let renderEpisodeSelectorList = () => {
+  let select = document.getElementById("select-menu");
+  let episodeTitle = document.getElementById("episode-title-container");
+
+  let defaultOption = document.createElement("option");
+  select.appendChild(defaultOption);
+  defaultOption.innerText = "Select an episode...";
+  defaultOption.selected; // technically not needed but added to be concise.
+
+  const allEpisodes = getAllEpisodes();
+  allEpisodes.forEach((episode) => {
+    let option = document.createElement("option");
+    select.appendChild(option);
+    let episodeCode = getEpisodeCode(episode);
+    option.value = episode.name;
+    option.innerText = `${episodeCode} - ${episode.name}`;
+  });
+
+  let selector = document.getElementById("select-menu");
+  selector.addEventListener("change", () => {
+    searchForSelectedEpisode(selector.value);
+    renderBackButton();
+    episodeTitle.innerText = `Game of Thrones`;
+  });
+};
+
+// Renders back navigation button and adds event listener
+let renderBackButton = () => {
+  let navigationContainer = document.getElementById("navigation-container");
+  navigationContainer.replaceChildren([]);
+
+  let button = document.createElement("button");
+  navigationContainer.appendChild(button);
+  button.id = "back-button";
```

Comment:
I'm not sure everything needs an id and a className. For styling, you can probably get away with just one.